docs/02-guides/public-configuration.mdx

```diff
@@ -0,0 +1,90 @@
+---
+title: Public configurations
+description:
+  The public configuration is a set of configuration options that are available
+  to any user of the application, they are accessible both from the client and
+  the server.
+sidebar_position: 5
+---
+
+<p>{frontMatter.description}</p>
+
+## Accessing Public Configuration
+
+To access the public configuration from the client you can use the
+[`usePublicConfig`](/docs/3.x/api-reference/front-commerce-core/react#usepublicconfig)
+hook, or the
+[`getPublicConfig`](/docs/3.x/api-reference/front-commerce-core/react#getpublicconfig)
+function.
+
+- The
+  [`usePublicConfig`](/docs/3.x/api-reference/front-commerce-core/react#usepublicconfig)
+  hook is mainly used to access the public configuration from a React component.
+- The
+  [`getPublicConfig`](/docs/3.x/api-reference/front-commerce-core/react#getpublicconfig)
+  function is used to access the public configuration from a non-React
+  component.
+
+:::tip
+
+We additionally also attach the public configuration to the
+`window.__FRONT_COMMERCE__.publicConfig` object.
+
+:::
+
+## Extending Public Configuration
+
+To extend the public configuration please ensure you have read the
+[Inject configurations](/docs/3.x/guides/configuration) section.
+
+The public configuration can be extended by
+[injecting a config provider](/docs/3.x/guides/configuration) which exposes a
+`public` object in the schema, for example:
+
+```ts title="./my-extensions/acme-extension/configProvider.ts"
+export default {
+  name: "acme-config",
+  schema: () => {
+    return {
+      public: {
+        acmeValue: {
+          doc: "The public api key for ACME",
+          format: String,
+          default: undefined,
+          env: "FRONT_COMMERCE_WEB_ACME_PUBLIC_KEY",
+        },
+      },
+    };
+  },
+};
+```
+
+You can then add this to your
+[extension definition](/docs/3.x/api-reference/front-commerce-core/defineExtension):
+
+```ts title="./my-extensions/acme-extension/index.ts"
+import configProvider from "./configProvider";
+import { defineExtension } from "@front-commerce/core";
+
+export default defineExtension({
+  name: "acme",
+  theme: "./extensions/acme/theme",
+  // add-start
+  configuration: {
+    providers: [configProvider],
+  },
+  // add-end
+});
+```
+
+You should now be able to access the value from the client:
```

Comment:
**nitpick:** it's not related to extending the public configuration

**suggestion:** add a second section for reading the value

```suggestion
## Reading the custom public configuration

You should now be able to access the value from the client as any other configuration:
```